src/nvim/os/shell.c

```diff
@@ -63,6 +63,8 @@ char **shell_build_argv(const char *cmd, const char *extra_args)
 
   rv[i] = NULL;
 
+  assert(rv[0]);
```

Comment:
Sorry, that's not what needed.
We need to make sure that first returned pointer (which points to shell option value) is not a NULL pointer.
`i > 0` wouldn't imply that (Imagine returning {NULL, NULL}).
It can be written `assert(rv[0] != NULL)`, if you prefer, but we've been following the other convention elsewhere.